lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDiagnosticParticipant.java

```diff
@@ -38,11 +40,15 @@
 	private MavenProjectCache projectCache;
 	MavenPluginManager pluginManager;
 	private final RepositorySystemSession repoSession;
+	private final MavenSession mavenSession;
+	private final BuildPluginManager buildPluginManager;
 
-	public MavenDiagnosticParticipant(MavenProjectCache projectCache, MavenPluginManager pluginManager, RepositorySystemSession repoSession) {
+	public MavenDiagnosticParticipant(MavenProjectCache projectCache, MavenPluginManager pluginManager, RepositorySystemSession repoSession, MavenSession mavenSession, BuildPluginManager buildPluginManager) {
```

Comment:
@mickaelistria the constructor of these "participant" classes are getting kind of long, maybe we should wrap all these objects into a container class? It'll get instantiated in `MavenPlugin` and be passed to all the participants.

Reply:
Isn't the mavenSession expected to contain a reference to the RepositorySystemSession or others?

Reply:
It probably is! I'll have to verify this.